src/lib.rs

```diff
@@ -38,72 +35,29 @@ use project_file_walker::get_project_file_parallel_iterator;
 use searcher::get_searcher;
 use treesitter::maybe_get_query;
 
-struct MaybeInitializedCaptureIndex(AtomicU32);
-
-impl MaybeInitializedCaptureIndex {
-    const UNINITIALIZED: u32 = u32::MAX;
-    const FAILED: u32 = u32::MAX - 1;
-
-    fn mark_failed(&self) -> bool {
-        loop {
-            let existing_value = self.0.load(Ordering::Relaxed);
-            if existing_value == Self::FAILED {
-                return false;
-            }
-            let did_store = self.0.compare_exchange(
-                existing_value,
-                Self::FAILED,
-                Ordering::Relaxed,
-                Ordering::Relaxed,
-            );
-            if did_store.is_ok() {
-                return true;
-            }
-        }
-    }
+#[derive(Default)]
+struct CaptureIndex(OnceLock<Result<u32, ()>>);
```

Comment:
So basically the `CaptureIndex` internally stores a `OnceLock<Result<...>>` to "clue itself in" to the fact that calculating the capture index has already failed, but then since in that case we want to print an error message and immediately kill the process, the "public API" of `CaptureIndex` doesn't actually need to acknowledge the existence of a failure case (so `.get_or_init()` returns plain old `u32`)

It strikes me that if we end up supporting usage as a library/crate (vs just as a command-line executable) then presumably we'd want to return a `Result`/`Err` when they pass something invalid (instead of killing the process)? Which would then mean restructuring some of this code I guess?

Reply:
Ah yeah, that's a good point, but seems fine for now.